toml/src/test/kotlin/org/rust/toml/completion/LocalCargoTomlDependencyVersionCompletionTest.kt

```diff
@@ -28,13 +28,31 @@ class LocalCargoTomlDependencyVersionCompletionTest : LocalCargoTomlCompletionTe
 
     fun `test complete sorted by semver versions`() = completeBasic(
         """
-        [dependencies]
-        foo = "<caret>"
+            [dependencies]
+            foo = "<caret>"
         """,
         listOf("0.4.7", "0.4.6", "0.4.5", "0.4.4", "0.4.3", "0.4.2", "0.4.1", "0.4.0", "0.4.0-rc.2", "0.4.0-rc.1", "0.4.0-rc", "0.3.17", "0.3.16", "0.3.15", "0.3.14", "0.3.13", "0.3.12", "0.3.11", "0.3.10", "0.3.9", "0.3.8", "0.3.7", "0.3.6", "0.3.5", "0.3.4", "0.3.3", "0.3.2", "0.3.1", "0.3.0", "0.2.11", "0.2.10", "0.2.9", "0.2.8", "0.2.7", "0.2.6", "0.2.5", "0.2.4", "0.2.3", "0.2.2", "0.2.1", "0.2.0", "0.1.6", "0.1.5", "0.1.4", "0.1.3", "0.1.2", "0.1.1", "0.1.0"),
         "foo" to CargoRegistryCrate.of("0.1.4", "0.4.3", "0.3.0", "0.1.1", "0.4.0", "0.3.5", "0.4.0-rc.1", "0.1.0", "0.4.2", "0.3.3", "0.3.4", "0.2.0", "0.1.3", "0.2.5", "0.3.8", "0.2.6", "0.4.6", "0.1.5", "0.2.11", "0.2.2", "0.4.5", "0.2.1", "0.3.1", "0.3.16", "0.3.6", "0.2.10", "0.4.4", "0.3.12", "0.2.9", "0.3.10", "0.1.6", "0.3.14", "0.2.4", "0.3.2", "0.1.2", "0.3.17", "0.3.9", "0.4.0-rc.2", "0.4.1", "0.3.15", "0.3.7", "0.4.0-rc", "0.2.7", "0.2.3", "0.4.7", "0.3.11", "0.2.8", "0.3.13"),
     )
 
+    fun `test complete sorted order minor, major and patch sections`() = completeBasic("""
+        [dependencies]
+        foo = "<caret>"
+    """,
+        // source: semver spec (https://semver.org/)
+        listOf("2.1.1", "2.1.0", "2.0.0", "1.0.0"),
+        "foo" to CargoRegistryCrate.of("1.0.0", "2.1.0", "2.1.1", "2.0.0")
```

Comment:
This order seems weird, right? From `semver.org`:

```
Precedence is determined by the first difference when comparing each of these identifiers from left to right as follows: Major, minor, and patch versions are always compared numerically.

Example: 1.0.0 < 2.0.0 < 2.1.0 < 2.1.1.
```

Reply:
Oh, sorry, that's my bad, I was looking at the bottom row instead of the top row. This ordering is fine.